src/Microsoft.ML.ImageAnalytics/ImageType.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Drawing;
+using Microsoft.ML.Runtime.Data;
+
+namespace Microsoft.ML.Runtime.ImageAnalytics
+{
+    public sealed class ImageType : StructuredType
+    {
+        public readonly int Height;
+        public readonly int Width;
+        public ImageType(int height, int width)
+           : base(typeof(Bitmap))
+        {
+            Contracts.CheckParam(height > 0, nameof(height));
+            Contracts.CheckParam(width > 0, nameof(width));
+            Contracts.CheckParam((long)height * width <= int.MaxValue / 4, nameof(height), $"{nameof(height)} * {nameof(width)} is too large");
```

Comment:
>$"{nameof(height)} * {nameof(width)} is too large" [](start = 91, length = 50)

If you really want it as format strings I'd encourage you to make this follow the `if (...) throw Contracts.Except` pattern. Or you could continue to make it a string literal by making this `nameof(height) + " * " + nameof(width) + " is too large"` or something.